core/schema/container.graphqls

```diff
@@ -61,121 +71,359 @@ type Container {
+  """
+  Retrieves this container plus a secret mounted into a file at the given path.
+  """
+  withMountedSecret(
+    """
+    Location of the secret file (e.g., "/tmp/secret.txt").
+    """
+    path: String!,
 
-  "Retrieves this container after unmounting everything at the given path."
-  withoutMount(path: String!): Container!
+    "Identifier of the secret to mount."
+    source: SecretID!
+  ): Container!
 
-  "Retrieves this container plus the contents of the given file copied to the given path."
-  withFile(path: String!, source: FileID!, permissions: Int): Container!
+  """
+  Retrieves this container after unmounting everything at the given path.
+  """
+  withoutMount(
+    """
+    Location of the cache directory (e.g., "/cache/node_modules").
+    """
+    path: String!
+  ): Container!
 
-  "Retrieves this container plus a new file written at the given path."
-  withNewFile(path: String!, contents: String, permissions: Int): Container!
+  """
+  Retrieves this container plus the contents of the given file copied to the given path.
+  """
+  withFile(
+    """
+    Location of the copied file (e.g., "/tmp/file.txt").
+    """
+    path: String!,
+
+    "Identifier of the file to copy."
+    source: FileID!,
+
+    """
+    Permission given to the copied file (e.g., 0600).
+
+    Default: 0644.
+    """
+    permissions: Int
+  ): Container!
+
+  """
+  Retrieves this container plus a new file written at the given path.
+  """
+  withNewFile(
+    """
+    Location of the written file (e.g., "/tmp/file.txt").
+    """
+    path: String!,
+
+    """
+    Content of the file to write (e.g., "Hello world!").
+    """
+    contents: String,
+
+    """
+    Permission given to the written file (e.g., 0600).
 
-  "Retrieves this container plus a directory written at the given path."
+    Default: 0644.
+    """
+    permissions: Int
+  ): Container!
+
+  """
+  Retrieves this container plus a directory written at the given path.
+  """
   withDirectory(
-    path: String!
-    directory: DirectoryID!
-    exclude: [String!]
+    """
+    Location of the written directory (e.g., "/tmp/directory").
+    """
+    path: String!,
+
+    "Identifier of the directory to write"
+    directory: DirectoryID!,
+
+    """
+    Patterns to exclude in the written directory (e.g., ["node_modules/**", ".gitignore", ".git/"]).
+    """
+    exclude: [String!],
+
+    """
+    Patterns to include in the written directory (e.g., ["*.go", "go.mod", "go.sum").
+    """
     include: [String!]
   ): Container!
 
-  "Retrieves this container plus a socket forwarded to the given Unix socket path."
-  withUnixSocket(path: String!, source: SocketID!): Container!
+  """
+  Retrieves this container plus a socket forwarded to the given Unix socket path.
+  """
+  withUnixSocket(
+    """
+    Location of the forwarded Unix socket (e.g., "/tmp/socket").
+    """
+    path: String!,
+
+    "Identifier of the socket to forward."
```

Comment:
```suggestion
    """
    Identifier of the socket to forward."
    """
```

Reply:
Where does not? I can't see the error here

Reply:
I just think it's better to always use the triple quotes instead of when you have a quote inside the description like in the examples. So it becomes the convention to always use `"""` for these descriptions.

Reply:
As it will improving diffing if we switch from 1-line to multi-lines description, I can agree with it too :)
Let's just use `"""` for those GraphQL description.

Reply:
> As it will improving diffing

Exactly, I mentioned it in https://github.com/dagger/dagger/pull/4367#issuecomment-1398350547.